lib/assets/javascripts/_modules/in-page-navigation.js

```diff
@@ -47,15 +47,15 @@
         fragment = fragmentForFirstElementInView()
       }
 
-      handleChangeInActiveItem(fragment)
+      handleChangeInActiveItem(fragment, true)
     }
 
     function handleScrollEvent () {
       handleChangeInActiveItem(fragmentForFirstElementInView())
     }
```

Comment:
Given that `handleChangeInActiveItem` now only does 1 thing when `skipHistory` is true, what do you think about re-factoring this code to remove `handleChangeInActiveItem` entirely? I think this might make the intent clearer and avoids the need to introduce a new boolean flag.

1. Within `handleInitialLoadEvent`, call `highlightActiveItemInToc` directly.
2. Within `handleScrollEvent`, call both `storeCurrentPositionInHistoryApi` and `highlightActiveItemInToc`, passing the fragment for the first element in view.
3. Remove the `handleChangeInActiveItem` function (not included in suggestion)

```suggestion
      highlightActiveItemInToc(fragment)
    }

    function handleScrollEvent () {
      var fragment = fragmentForFirstElementInView()
      storeCurrentPositionInHistoryApi(fragment)
      highlightActiveItemInToc(fragment)
    }
```

What do you reckon?

Reply:
👍🏻  That makes sense!